cli/orchestration/outputs-sharing.md

```diff
@@ -0,0 +1,182 @@
+- `mock` _(optional, any)_ - A mock value for the case when the dependent stack outputs are not applied (used for previewing the planned changes).
+
+When `terramate generate` is used with the example configuration above, it generates
+the code below:
+```
+// TERRAMATE: GENERATED AUTOMATICALLY DO NOT EDIT
+
+variable "vpc_id" {
+  type = string
+}
+```
+
+For the same example, let's say executing `terraform output -json` in the `vpc` stack gives the output below:
+
+```json
+{
+  "vpc_id": {
+    "description": "The AWS VPC id",
+    "value": "<value>",
+    "sensitive": false
+  }
+}
+```
+
+then the expression `outputs.vpc_id.value` (from the `input.value`) returns the `"<value>"` string.
+
+# Configure stack outputs
+
+The `output` block defines what values are exported from the stack. Those values
+can be used by other stacks defining `input` blocks.
+See example below:
+
+```
+output "vpc_id" {
+  backend   = "default"
+  value     = module.vpc.id
+  sensitive = false
+}
+```
+
+The block label is required and it defines the name of the output.
+It has the attributes below:
+
+- `backend` _(required, string)_ - The name of the sharing backend that this input refers to.
+- `value` _(required, expression)_ - The expression that exports the stack's resource.
+- `sensitive` _(optional, boolean)_ - If the value is sensitive or not.
+
+# Orchestration
+
+During orchestration, if sharing outputs is turned on, Terramate will invoke the
+`sharing_backend.command` on demand to gather outputs of referenced stacks and,
+depending on the `sharing_backend.type`, supply the stack inputs.
+The feature can be turned on with `terramate run --enable-sharing ...` or in the
+case of scripts, by setting it as a command object, see example below:
+
+```hcl
+script "terraform" "deploy" {
+  job {
+    name        = "Terraform Apply"
+    description = "Initialize, validate, plan, and apply Terraform changes."
+    commands = [
+      ["terraform", "init", "-lock-timeout=5m"],
+      ["terraform", "validate"],
+      ["terraform", "plan", "-out", "out.tfplan", "-lock=false", {
+        enable_sharing = true
+      }],
+      ["terraform", "apply", "-input=false", "-auto-approve", "-lock-timeout=5m", "out.tfplan", {
+        cloud_sync_deployment          = true
+        cloud_sync_terraform_plan_file = "out.tfplan"
+      }],
+    ]
+  }
+}
+```
+
+# Mocking
+
+If your team makes use of code review, very often both the dependendy and the 
+dependent stacks change in the same PR and in this case, the plan of the 
+dependent stack cannot retrieve the outputs of the dependency because
+they were not applied yet. For this case, the `input` block supports `mock`
+attribute and its value will be used in the case of the dependency stack fail
+to return the outputs, or the referenced output is not applied in the plan.
+The mock value is only used if `--mock-on-fail` is provided to `terramate run` or
+`mock_on_fail` option is provided to the script command.
```

Comment:
```suggestion
If your team uses code reviews, it's common for both the dependency and dependent stacks to change in the same PR. In such cases, the plan of the dependent stack may fail to retrieve the outputs from the dependency because they haven't been applied yet. To handle this, the `input` block supports a `mock` attribute, which provides a fallback value if the dependency stack fails to return the outputs or if the referenced output isn't applied in the plan. The mock value is used only when `--mock-on-fail` is provided to `terramate run` or when the `mock_on_fail` option is included in the script command.
```